### ISSUE TITLE
Need some advice on using Pickler (java TimeZone)

### ISSUE THREAD

**Issue by jenwirth** [softwaremill/tapir#3287]
I am attempting to transition from Circe to Pickler. The main motivation is to eliminate the challenge of keeping the schema and JSON aligned, an issue that has led to discrepancies in the past. Some partners even pointed out that the schema definitions were inconsistent with the swagger documentation. 

However, we're facing a seemlingly basic initial hurdle, specifically needing a `Pickler[TimeZone]`.

```scala
import upickle.default.{ReadWriter, readwriter}
import sttp.tapir.json.pickler.*
import java.util.TimeZone
import sttp.tapir.*

given Schema[TimeZone] = Schema(SchemaType.SString())
given ReadWriter[TimeZone] = readwriter[String].bimap[TimeZone](_.getID, TimeZone.getTimeZone)
given Pickler[TimeZone] = Pickler.nonMirrorPickler
```

The error:

```
|given Pickler[TimeZone] = Pickler.nonMirrorPickler
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^
    |No given instance of type Reader[java.util.TimeZone] was found.
    |I found:
    |
    |    this.superTypeReader[java.util.TimeZone, V](
    |      /* missing */summon[deriving.Mirror.ProductOf[java.util.TimeZone]], ???, ???,
    |      ???)
    |
    |But Failed to synthesize an instance of type deriving.Mirror.ProductOf[java.util.TimeZone]: class TimeZone is not a generic product because it is not a case class.

```

Why does it look for a `Mirror`?

**Comment by kciesielski** [comment]
@jenwirth The `nonMirrorPickler` method shouldn't be visible outside tapir. 
Your case helped me to indentify issues with the public API, which has to be fixed.
After the fix, the proper way of constructing a Pickler for such a type would be
```scala

given Schema[TimeZone] = Schema(SchemaType.SString())
given ReadWriter[TimeZone] = readwriter[String].bimap[TimeZone](_.getID, TimeZone.getTimeZone)
given Pickler[TimeZone] = Pickler.derived
```

I'll publish a PR with updates later today. I'll think of a workaround as well.

**Comment by jenwirth** [comment]
Thanks (and you're welcome :)

Nonetheless this still looks for a mirror (it was the reason i tried using `nonMirrorPickler`:


```
scala> import upickle.default.{ReadWriter, readwriter}
     | import sttp.tapir.json.pickler.*
     | import java.util.TimeZone
     | import sttp.tapir.*
                                                                                                                                                                                                                                            
scala> given Schema[TimeZone] = Schema(SchemaType.SString())
     | given ReadWriter[TimeZone] = readwriter[String].bimap[TimeZone](_.getID, TimeZone.getTimeZone)
     | given Pickler[TimeZone] = Pickler.derived
-- [E172] Type Error: ----------------------------------------------------------
3 |given Pickler[TimeZone] = Pickler.derived
  |                                         ^
  |No given instance of type deriving.Mirror.Of[java.util.TimeZone] was found for parameter x$2 of method derived in object Pickler. Failed to synthesize an instance of type deriving.Mirror.Of[java.util.TimeZone]:
  |     * class TimeZone is not a generic product because it is not a case class
  |     * class TimeZone is not a generic sum because it is not a sealed class
1 error found

```

**Comment by jenwirth** [comment]
To continue the process of the POC, i am using this hack

```scala
  given Pickler[TimeZone] = Pickler(
    new TapirPickle[TimeZone] {
      override lazy val reader = reader[String].map(TimeZone.getTimeZone)
      override lazy val writer = writer[String].comap(_.getID)
    },
    summonInline[Schema[TimeZone]]
  )

```

**Comment by kciesielski** [comment]
Yes, using constructor directly should work for now :+1: 
Regarding the question why compiler complains about missing mirror when calling `nonMirrorPickler` - first of all, this method was not intended as public API, which is fixed together with the actual issue in #3289. This method calls implicit resolution on uPickle types, which may somehow run into uPickles `superTypeReader`, which then tries to resolve the mirror. This should not happen in situations when it's used internally.

With the fix, I hope using just `Pickler.derive` should do the trick when readers/writers and schema are provided.
